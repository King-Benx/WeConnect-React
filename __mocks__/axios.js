const axiosMock = jest.genMockFromModule('axios');
let mockResponse = {
    data:{
        results:{
            businesses: [
                {
                  "category": "Information Technology",
                  "created_by": "admin",
                  "date_created": "Tue, 10 Jul 2018 12:27:46 GMT",
                  "description": "business 1 description",
                  "id": 1,
                  "last_modified": "Tue, 10 Jul 2018 12:27:46 GMT",
                  "location": "Kampala",
                  "name": "business 1",
                  "user_id": 1
                },
                {
                    "category": "Information Technology",
                    "created_by": "admin",
                    "date_created": "Tue, 10 Jul 2018 12:27:46 GMT",
                    "description": "business 1 description",
                    "id": 1,
                    "last_modified": "Tue, 10 Jul 2018 12:27:46 GMT",
                    "location": "Kampala",
                    "name": "business 1",
                    "user_id": 1
                  },
            ]
        }
    }
}
axiosMock.get.mockImplementation(req);
function req(){
   return new Promise((resolve)=>{
       axiosMock.delayTimer = setTimeout(() => {
           resolve(mockResponse)
       }, 1000);
   })   
}
module.exports = axiosMock;