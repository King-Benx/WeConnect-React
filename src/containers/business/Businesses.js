import React from 'react';
import { Redirect } from 'react-router-dom';
import { Button, Table } from 'react-bootstrap';
import superagent from 'superagent';
import { BASE_URL } from '../../custom/constants';
import CustomFunctions from '../../custom/CustomFunctions';
import DashboardNavigation from '../../navigation/DashboardNavigation';

class Businesses extends React.Component{
    
    constructor(){
        super();
        this.state = {
            all_businesses:[],
            page:'',
            limit:'',
            number_of_records: ''
        }
    }
    
    showReviews = (event) => {
        this.props.history.push('/all_businesses/'+event.target.id+'/reviews');
    }

    componentDidMount(){
        superagent
        .get(BASE_URL+'api/v1/businesses?page='+this.state.page+'&limit='+this.state.limit)
        .set({'x-access-token':CustomFunctions.getToken()})
        .end((err,res)=>{
            if(err){
                CustomFunctions.createNotifications(err.status,res.body.message)
            };
            if (res.status === 404){
                this.setState({all_businesses:[]});
            }else if(res.status === 200){
                console.log(res.body.results.businesses.length)
                this.setState({all_businesses : res.body.results.businesses});
            }
        });

        // superagent
        // .get(BASE_URL+'api/v1/businesses')
        // .set({'x-access-token':CustomFunctions.getToken()})
        // .end((err,res)=>{
        //     this.setState({number_of_records : res.body.results.businesses.length});
        // });

    }

    // handlePageLimit = (event) => {
    //     this.setState({limit: event.target.id, page_numbers: Math.ceil(this.state.number_of_records/ Number(event.target.id))})
    //     this.componentDidMount()
    // }

    // handlePageChange = (event) => {
    //     this.setState({ page: event.target.value, limit:this.state.limit })
    //     this.componentDidMount()
    // }

    render(){
        const table_data = this.state.all_businesses.map(((business)=>{
            return(
                <tr key={business.id.toString()}>
                    <td>{business.date_created}</td>
                    <td>{business.name}</td>
                    <td>{business.location}</td>
                    <td>{business.category}</td>
                    <td>{business.description}</td>
                    <td>
                        <Button id={business.id} onClick={this.showReviews} bsStyle="info" block>
                            <i className="glyphicon glyphicon-info-sign"></i> View
                        </Button>
                    </td>
                </tr>
            );
        }));

        // const iterator = [];
        // if (this.state.limit !== 0){
        //     for (let i=1; i<= this.state.page_numbers; i ++){
        //         iterator.push(i);
        //     }
        // }
        // else{
        //     iterator.push(1);
        // }

        // const pagination_data = iterator.map((number) => {
        //     return(
        //         <Pagination.Item key={number} onClick={ this.handlePageChange } value={ number }>{ number }</Pagination.Item>
        //     );
        // });

        return (
            CustomFunctions.checkAuthentication() ? (   
            <div className="row"> 
                <DashboardNavigation/>
                <div className="col-sm-9 content-wrapper">
                    <div className="panel-heading">
                        <h3 className="panel-title">
                            <b>All Registered Businesses</b>
                        </h3>
                    </div>
                    <div className="panel panel-default">
                        <div className="panel-body">
                            {/* <div className="row text-right">
                                <input type="number" min="1" value={this.state.limit } onChange={ this.handlePageLimit } />
                            </div>
                            <br/> */}
                            <div className="clearfix"></div>
                            <div className="table-responsive">
                                <Table striped bordered condensed hover>
                                    <thead>
                                        <tr>
                                            <th>Date of Creation</th>
                                            <th>Name</th>
                                            <th>Location</th>
                                            <th>Category</th>
                                            <th>Description</th>
                                            <th>View</th>
                                        </tr>
                                    </thead>
                                    <tbody>
                                        {table_data}
                                    </tbody>
                                </Table>
                            </div>
                            {/* <div className="row text-center">
                                <Pagination>
                                    <Pagination.Prev />
                                        { pagination_data } 
                                    <Pagination.Next />
                                </Pagination>
                            </div> */}
                        </div>
                    </div>
                </div>
            </div>
            ): <Redirect to={{pathname:'/login'}}/>
        );
    }
}
export default Businesses;