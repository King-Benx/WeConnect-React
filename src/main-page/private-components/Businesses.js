import React from 'react';
import {Redirect} from 'react-router-dom';
import superagent from 'superagent';
import { BASE_URL } from '../functional-resources/urls';
import checkAuthentication from '../functional-resources/checkAuthentication';
import DashboardNavigation from '../shared-components/DashboardNavigation';

class Businesses extends React.Component{
    constructor(){
        super();
        this.state = {
            all_businesses:[]
        }
    }
    componentDidMount(){
        superagent
        .get(BASE_URL+'api/v1/businesses')
        .set({'x-access-token':localStorage.getItem('token')})
        .end((err,res)=>{
            if(err){alert(res.body.message)};
            if (res.status === 404){
                this.setState({all_businesses:[]});
            }else if(res.status === 200){
                this.setState({all_businesses:res.body.results.businesses});
            }
        });
    }
    render(){
        const table_data= this.state.all_businesses.map(((business)=>{
            return(
                <tr key={business.id.toString()}>
                    <td>{business.date_created}</td>
                    <td>{business.name}</td>
                    <td>{business.location}</td>
                    <td>{business.category}</td>
                    <td>{business.description}</td>
                    <td>
                        <a href="reviews.html">10 reviews</a>
                    </td>
                </tr>
            );
        }));
        return (
            checkAuthentication()? (   
            <div className="row"> 
                <DashboardNavigation/>
                <div className="col-sm-9 content-wrapper">
                    <div className="panel-heading">
                        <h3 className="panel-title">
                            <b>All Registered Businesses</b>
                        </h3>
                    </div>
                    <div className="panel panel-default">
                        <div className="panel-body">
                            <div className="table-responsive">
                                <table className="table table-bordered table-checked">
                                    <thead>
                                        <tr>
                                            <th>Date of Creation</th>
                                            <th>Name</th>
                                            <th>Location</th>
                                            <th>Category</th>
                                            <th>Description</th>
                                            <th>Reviews</th>
                                        </tr>
                                    </thead>
                                    <tbody>
                                        {table_data}
                                    </tbody>
                                </table>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
            ): <Redirect to={{pathname:'/login'}}/>
        );
    }
}
export default Businesses;