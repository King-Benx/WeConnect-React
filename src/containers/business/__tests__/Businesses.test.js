import React from 'react';
import { shallow } from 'enzyme';
import Businesses from "../Businesses";
import renderer from 'react-test-renderer';
import axios from '../../../../__mocks__/axios'

describe('These are tests for the businesses page ', () =>{

    let mountedBusinesses;
    beforeEach(() => {
        mountedBusinesses = shallow(<Businesses />)
    });

    it('renders without crashing', () => {
       shallow(<Businesses />)
       const tree = renderer.create(<Businesses />).toJSON();
       expect(tree).toMatchSnapshot();
    });

    it('component did mount', () => {
       mountedBusinesses.instance().componentDidMount()
    })

    // it('calls axios.get in #componentDidMount', () => {
    //     return mountedBusinesses.instance().componentDidMount().then(()=>{
    //         expect(axios.get).toHaveBeenCalled()
    //     })
    // })

    it('has a dashboard', () => {
        const dashboards = mountedBusinesses.find('DashboardNavigation');
        expect(dashboards.length).toBe(1)
     })

     it('has a table', () => {
        const tables = mountedBusinesses.find('Table');
        expect(tables.length).toBe(1)
     })

     it('has divs', () => {
        const divs = mountedBusinesses.find('div');
        expect(divs.length).toBe(11)
     })

     it('has h3', () => {
        const h3s = mountedBusinesses.find('h3');
        expect(h3s.length).toBe(1)
     })

     it('has b', () => {
        const bs = mountedBusinesses.find('b');
        expect(bs.length).toBe(1)
     })

});