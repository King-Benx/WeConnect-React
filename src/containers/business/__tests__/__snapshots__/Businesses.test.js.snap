// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`These are tests for the businesses page  renders without crashing 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-9 content-wrapper"
  >
    <div
      className="panel-heading"
    >
      <h3
        className="panel-title"
      >
        <b>
          All Registered Businesses
        </b>
      </h3>
    </div>
    <div
      className="panel panel-default"
    >
      <div
        className="panel-body"
      >
        <div
          className="row text-right"
        >
          <div
            className="col-xs-6"
          />
          <div
            className="col-xs-6 "
          >
            <select
              className="form-control"
              id="limit"
              onChange={[Function]}
            >
              <option>
                Limit records per page
              </option>
              <option
                value="1"
              >
                1
              </option>
              <option
                value="2"
              >
                2
              </option>
              <option
                value="3"
              >
                3
              </option>
            </select>
          </div>
        </div>
        <br />
        <div
          className="clearfix"
        />
        <div
          className="table-responsive"
        >
          <table
            className="table table-striped table-bordered table-condensed table-hover"
          >
            <thead>
              <tr>
                <th>
                  Date of Creation
                </th>
                <th>
                  Name
                </th>
                <th>
                  Location
                </th>
                <th>
                  Category
                </th>
                <th>
                  Description
                </th>
                <th>
                  View
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
        <div
          className="row text-center"
        >
          <ul
            className="pagination"
          >
            <li
              aria-label="Prev"
              className=""
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                ‹
              </a>
            </li>
            <li
              aria-label="Next"
              className=""
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                ›
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
